src/main/java/ru/ralsei/whatcanyousee/gameactivity/GooglePlayHandler.java

```diff
@@ -0,0 +1,641 @@
+                    }
+                })
+                .addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        activity.handleException(e, activity.getString(R.string.leaderboards_exception));
+                    }
+                });
+    }
+
+    /**
+     * Saves player's achievements and statistic to the cloud.
+     */
+    void pushAccomplishments() {
+        if (GoogleSignIn.getLastSignedInAccount(activity) == null) {
+            return;
+        }
+
+        if (activity.getGameStatistic().getMazeGameTime() != -1) {
+            mLeaderboardsClient.submitScore(activity.getString(R.string.leaderboard_maze_game_best_time), activity.getGameStatistic().getMazeGameTime());
+        }
+
+        if (activity.getGameStatistic().getCodeGameTime() != -1) {
+            mLeaderboardsClient.submitScore(activity.getString(R.string.leaderboard_code_game_best_time), activity.getGameStatistic().getCodeGameTime());
+        }
+
+        if (activity.getGameStatistic().getLeverGameTime() != -1) {
+            mLeaderboardsClient.submitScore(activity.getString(R.string.leaderboard_lever_game_best_time), activity.getGameStatistic().getLeverGameTime());
+        }
+
+        if (activity.getGameStatistic().getCodeGameMistakeTaken() != -1) {
+            mLeaderboardsClient.submitScore(activity.getString(R.string.leaderboard_code_game_least_mistake_taken), activity.getGameStatistic().getCodeGameMistakeTaken());
+        }
+
+    }
+
+    /**
+     * Get (or remove) teammate's Participant account.
+     */
+    private void updateRoom(Room room) {
+        if (room == null) {
+            return;
+        }
+
+        Participant[] participants = new Participant[2];
+        participants = room.getParticipants().toArray(participants);
+
+        myParticipant = null;
+        teammateParticipant = null;
+
+        for (Participant participant : participants) {
+            if (participant.getParticipantId().equals(mMyId)) {
+                myParticipant = participant;
+            } else {
+                teammateParticipant = participant;
+            }
+        }
+    }
+
+    String getIncomingInvitationId() {
+        return mIncomingInvitationId;
+    }
+
+    GoogleSignInClient getGoogleSignInClient() {
+        return mGoogleSignInClient;
+    }
+
+    void setGoogleSignInClient(GoogleSignInClient client) {
+        this.mGoogleSignInClient = client;
+    }
+
+    InvitationsClient getInvitationsClient() {
+        return mInvitationsClient;
+    }
+
+    InvitationCallback getInvitationCallback() {
+        return mInvitationCallback;
+    }
+
+    RealTimeMultiplayerClient getRealTimeMultiplayerClient() {
+        return mRealTimeMultiplayerClient;
+    }
+
+    void setIncomingInvitationId(String id) {
+        mIncomingInvitationId = null;
+    }
+
+    String getRoomId() {
+        return mRoomId;
```

Comment:
In general, this "m" prefix is not a vary Java thing. It's more often used in C++. But if it's used, use it for all class members, keep the code consistent.